src/applications/mhv-secure-messaging/components/MessageThread/MessageThreadItem.jsx

```diff
@@ -103,6 +105,14 @@ const MessageThreadItem = props => {
           ? `expand-message-button-for-print-${messageId}`
           : `expand-message-button-${messageId}`
       }
+      onClick={e => {
+        setIsItemExpanded(e.target?.getAttribute('open'));
+      }}
+      data-dd-action-name={`${
+        isitemExpanded
+          ? 'Accordion Expanded Message'
+          : 'Accordion Collapsed Message'
+      }`}

```

Comment:
What do you think about calling `getAttribute` in `data-dd-action-name` directly to potentially avoid unnecessary rerenders?
```suggestion
      data-dd-action-name={`${
        e.target?.getAttribute('open')
          ? 'Accordion Expanded Message'
          : 'Accordion Collapsed Message'
      }`}

```